### ISSUE TITLE
🐛 [cloud_firestore] Settings.CACHE_SIZE_UNLIMITED stop working on iOS after upgrade Flutter to 3.10.5

### ISSUE THREAD

**Issue by muriloventuroso** [firebase/flutterfire#11172]
## Bug report

After updating Flutter to version 3.10.5 setting cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED stopped working, showing the error when running the app:

```
10.10.0 - [FirebaseFirestore][I-FST000001] Invalid argument: Cache size must be set to at least 1048576 bytes
libc++abi: terminating due to uncaught exception of type std::invalid_argument: Cache size must be set to at least 1048576 bytes
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001f5782558 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`:
->  0x1f5782558 <+8>:  b.lo   0x1f5782578               ; <+40>
    0x1f578255c <+12>: pacibsp
    0x1f5782560 <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1f5782564 <+20>: mov    x29, sp
Target 0: (Runner) stopped.
```

## Sample code

```
FirebaseFirestore.instance.settings = const Settings(
            persistenceEnabled: true,
            sslEnabled: true,
            cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED,
          );
```

It works if I set a specific value for the cache size:

```
FirebaseFirestore.instance.settings = const Settings(
            persistenceEnabled: true,
            sslEnabled: true,
            cacheSizeBytes: 1048576 * 10000,
          );
```
---

Only on IOS.

### Flutter doctor
<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.4 22F66 darwin-arm64, locale pt-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.79.0)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```

</details>

---

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
cloud_firestore: ^4.8.1
firebase_core: ^2.14.0
```

</details>

---

**Comment by darshankawar** [comment]
Thanks for the report @muriloventuroso 
I tried the code sample using latest stable version and running on iOS simulator with which I didn't get the crash as reported. The app compiled and built properly:

<img width="354" alt="Screenshot 2023-06-22 at 3 28 33 PM" src="https://github.com/firebase/flutterfire/assets/67046386/a2f50d01-33a9-4e8b-9723-1f81ebb22b82">

```
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
  FirebaseFirestore.instance.settings = Settings(persistenceEnabled: true, sslEnabled: true, cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED);

  if (shouldUseFirestoreEmulator) {
    FirebaseFirestore.instance.useFirestoreEmulator('localhost', 8080);
  }
  runApp(FirestoreExampleApp());
}
```

Does it occur using simulator or physical device ?

Also, if you comment out the persistence code line, does it run properly ?

There's an open issue related to persistence causing app crash: https://github.com/firebase/flutterfire/issues/11138

**Comment by darshankawar** [comment]
@muriloventuroso 
The issue related to `persistence` has been fixed as part of https://github.com/firebase/flutterfire/pull/11174 so once new plugin version with the fix is available, use it and run your scenario again to see if the error is fixed at your end or not.

**Comment by muriloventuroso** [comment]
It is fixed in cloud_firestore version 4.8.2
